xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/DecimalField.java

```diff
@@ -49,29 +51,47 @@ public void setValue(Object value) throws InvalidOutcomeException {
 
         super.setData(value.toString());
     }
-    
+
     @Override
     public void setNgDynamicFormsValidators(JSONObject validators) {
-        Logger.msg("***************setNgDynamicFormsValidators***************");
+        //WE shouzld be using locale sepcific separators
+        //char separator = new DecimalFormatSymbols(Locale.getDefault(Locale.Category.FORMAT)).getDecimalSeparator();
+        char separator = '.';
+
         super.setNgDynamicFormsValidators(validators);
-        Logger.msg("***************" + name + " PATTERN START***************");
 
         if (StringUtils.isNotBlank(pattern)) {
             validators.put("pattern", pattern);
-        } else if (contentType.hasFacet(Facet.TOTALDIGITS) || contentType.hasFacet(Facet.FRACTIONDIGITS)) {
-            String pattern = buildPatternFromRestrictions();
-            validators.put("pattern", pattern);
         }
-        
-        Logger.msg("***************" + name + " PATTERN END***************");
-    }
+        else {
+            Integer totalDigits = null;
+            Integer fractionDigits = null;
 
-    private String buildPatternFromRestrictions() {
-      BigDecimal totalDigits = getRangeValue(Facet.TOTALDIGITS);
-      BigDecimal fractionDigits = getRangeValue(Facet.FRACTIONDIGITS);
-      String naturalNumbers = totalDigits.subtract(fractionDigits).toPlainString();
-      String decimals = fractionDigits.equals(BigDecimal.ZERO) ? "" : fractionDigits.toPlainString();
-      
-      return "^\\\\d{0," + naturalNumbers + "}\\\\.?\\\\d{0," + decimals + "}$";
+            if(contentType.hasFacet(Facet.TOTALDIGITS)) {
+                Facet f = contentType.getFacet(Facet.TOTALDIGITS);
+                totalDigits = Integer.valueOf(f.getValue());

```

Comment:
handle exceptions

Reply:
what exceptions?

Reply:
`public static Integer valueOf(String s) throws NumberFormatException`

Reply:
The is no checked exception in the API, runtime exception are handled at the top level of the call, mainly in the restapi